doc/extensions/zephyr/application.py

```diff
@@ -115,22 +120,25 @@ def run(self):
             'gen_args': gen_args,
             'build_args': build_args,
             'source_dir': source_dir,
-            'goals': goals
+            'goals': goals,
+            'compact': compact
             }
 
         # Build the command content as a list, then convert to string.
         content = []
 
         if zephyr_app:
             content.append('$ cd $ZEPHYR_BASE/{}'.format(zephyr_app))
-            content.append('')
+            if not compact:
+                content.append('')
         elif app:
             content.append('$ cd {}'.format(app))
-            content.append('')
+            if not compact:
+                content.append('')
 
         if skip_config:
             content.append("# If you already made a build directory ({}) and ran cmake, just 'cd {}' instead.".format(build_dir, build_dir))  # noqa: E501
-        else:
+        elif not compact:
```

Comment:
This doesn't behave as documented: if `:maybe-skip-config:` and `:compact:` are both set, the comment generated by maybe-skip-config is still present, even though compact says no comments will be present in the output text.

It seems that an error should be raised if `skip_config and compact`, as  they can't both win, and the behavior should be adjusted so the one which is present (if either) takes effect.